app/models/observation_accuracy_experiment.rb

```diff
@@ -0,0 +1,559 @@
+      user_id = validator.user_id
+      groundtruth_taxa = Identification.
+        select( "DISTINCT ON ( observation_id ) observation_id, taxon_id, " \
+          "disagreement, previous_observation_taxon_id" ).
+        where( observation_id: obs_ids, user_id: user_id, current: true ).
+        order( "observation_id DESC, created_at DESC" ).
+        pluck( :observation_id, :taxon_id, :disagreement, :previous_observation_taxon_id ).
+        group_by( &:shift ).
+        transform_values do | values |
+          values.map {| v | { taxon_id: v[0], disagreement: v[1], previous_observation_taxon_id: v[2] } }
+        end
+      groundtruths[user_id] = groundtruth_taxa
+    end
+
+    responding_validators = groundtruths.values.reject( &:empty? ).count
+    validated_observations = groundtruths.values.map( &:keys ).flatten.uniq.count
+
+    samples.each do | sample |
+      oid = sample.observation_id
+      test_taxon = sample.taxon_id
+      matches = groundtruths.values.map {| groundtruth_taxa | groundtruth_taxa[oid] }.compact.flatten
+      if matches.empty?
+        sample.correct = nil
+      else
+        qualities_for_row = []
+        matches.each do | match |
+          groundtruth_taxon = match[:taxon_id]
+          disagreement = match[:disagreement]
+          previous_observation_taxon_id = match[:previous_observation_taxon_id]
+          quality = assess_quality( test_taxon, groundtruth_taxon, disagreement, previous_observation_taxon_id )
+          qualities_for_row << quality
+        end
+        sample.correct = if !qualities_for_row.map( &:nil? ).all? &&
+            qualities_for_row.compact.uniq.count > 1
+          # conflict
+          -1
+        else
+          qualities_for_row[0]
+        end
+      end
+      sample.save!
+    end
+
+    if samples.select {| q | q.correct == -1 }.count.positive?
+      obs_ids = qualities.select {| q | q.correct == -1 }.map( &:observation_id ).join( "," )
+      puts "There is at least one conflict"
+      puts "#{Site.default.url}/observations/identify?" \
+        "reviewed=any&quality_grade=needs_id%2Cresearch%2Ccasual&id=#{obs_ids} )"
+    else
+      quality_stats = get_quality_stats( samples.map( &:correct ) )
+      precision_stats = get_precisions_stats( samples.map {| sample | calculate_precision( sample.descendant_count ) } )
+      self.assessment_date = Time.now
+      self.responding_validators = responding_validators
+      self.validated_observations = validated_observations
+      self.low_acuracy_mean = quality_stats[0]
+      self.low_acuracy_variance = quality_stats[1]
+      self.high_accuracy_mean = quality_stats[2]
+      self.high_accuracy_variance = quality_stats[3]
+      self.precision_mean = precision_stats[0]
+      self.precision_variance = precision_stats[1]
+      save!
+    end
+  end
+
+  def get_sample_for_user_id( user_id )
+    return nil unless validator == ObservationAccuracyValidator.where( user_id: user_id ).first
```

Comment:
is validator defined somewhere? If not, did you intend to assign to validator with `=` instead of comparing with `==`?